server/issue.go

```diff
@@ -193,6 +193,78 @@ func httpAPIGetCreateIssueMetadata(ji Instance, w http.ResponseWriter, r *http.R
 	return http.StatusOK, nil
 }
 
+func (p *Plugin) assignJiraIssue(mmUserId, issueKey, jiraAssignee string) (*model.CommandResponse, error) {
+	ji, err := p.LoadCurrentJIRAInstance()
+	if err != nil {
+		return nil, err
+	}
+
+	jiraUser, err := ji.GetPlugin().LoadJIRAUser(ji, mmUserId)
+	if err != nil {
+		return nil, err
+	}
+
+	jiraClient, err := ji.GetJIRAClient(jiraUser)
+	if err != nil {
+		return nil, err
+	}
+
+	// check for valid issue key
+	_, _, err = jiraClient.Issue.Get(issueKey, nil)
+	if err != nil {
+		errorMsg := fmt.Sprintf("We couldn't find the issue key `%s`.  Please confirm the issue key and try again.", issueKey)
+		return responsef(errorMsg), nil
+	}
+
+	// Get list of assignable assignees
+	url := fmt.Sprintf("rest/api/2/user/assignable/search?issueKey=%s&query=%s", issueKey, jiraAssignee)
+	req, err := jiraClient.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := jiraClient.Do(req, nil)
+	if err != nil {
+		if res.Response.StatusCode == 401 {
+			errorMsg := fmt.Sprintf("You do not have the appropriate permissions to perform this action. Please contact your Jira administrator.")
+			return responsef(errorMsg), nil
+		}
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(res.Body)
```

Comment:
should have 
`defer res.Body.Close()` before the ReadAll (to prevent resource leaks)